### PR TITLE
[serve] immediately send ping in router when receiving new replica set

### LINKED ISSUE #47036: [serve] proxy should ping replica immediately after receiving new actor handle
When proxies get an updated set of running replicas, it receives the actor handles for them from the controller. However the actor handle itself is only a pointer that contains the actor ID, but doesn't actually contain all actor info such as the actor address. This info is populated by the GCS upon first request made through that actor handle.

If GCS goes down before the proxy gets to make that first request, requests will hang in the duration of GCS of failure. To improve fault tolerance, the router should ping the replica immediately after receiving a new actor handle to populate the actor info.

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

**Context**:
When a new set of `RunningReplicaInfos` are broadcasted to a router, the nested actor handles are "empty" and don't hold the necessary actor info (e.g. actor address) to send a request to that replica. Upon first request, the handle fetches that info from the GCS.

This can cause fault tolerance issues because if the GCS goes down immediately after a replica set change is broadcasted to a router, that router is unable to send requests to any replicas; they will all be blocked until the GCS recovers.

**Fix**:
- Upon receiving a new replica set, the router actively probes the queue lengths for each replica. This simultaneously sends an initial "ping" using the actor handle which populates the actor info from the GCS, and also updates the queue length cache.
- Since proxy sends its "self" actor handle to the replica for the replica to call `receive_asgi_messages`, also push this actor handle to replicas upon replica set change, otherwise proxy requests to new replicas will hang when GCS is down.

## Related issue number

closes https://github.com/ray-project/ray/issues/47036

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py
Comment (edoakes) [inline python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py:289]: hm can we do this only for the replicas that were added instead of all?

File: python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py
Comment (edoakes) [inline python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py:253]: let's add a method to the interface, shouldn't be accessing the `_actor_handle` private attribute

Comment (edoakes) [reply python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py:253]: that way it can be tested as well

File: python/ray/serve/_private/replica.py
Comment (GeneDer) [inline python/ray/serve/_private/replica.py:324]: Should we do something to the handle? Also maybe add a type hint is it's required 🙃

File: python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py
Comment (zcin) [reply python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py]: yes! for some reason I thought it would mess with the fault tolerance, but seems like the actor info is stored per-process not per actor handle. changed to only ping new replicas.

File: python/ray/serve/_private/replica.py
Comment (zcin) [reply python/ray/serve/_private/replica.py:324]: doing something with the handle seems unnecessary for now, I think if you pass any actor handle as an argument in a ray remote call like:
```
x.remote(actor_handle)
```
then ray core does some processing under the hood that requires making a call to the GCS, so if this `actor_handle` was never "pushed" to actor beforehand then this call hangs. "Pushing" it once is enough to unblock the call though when the GCS goes down.

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
fill in the "good comment"s before merging please :)

**Pull request merged**
